### PR TITLE
How to build discord bot with JavaScript

### PR DESCRIPTION
**Pull request by Kamau-ke**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

`This closes #issue-number`

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by Kamau-ke** [comment]
this fixes #5159

**Comment by LinusMuema** [comment]
Hi @Kamau-ke 

Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept. Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content. Be sure to see our resources page to see more info on plagiarism and what is considered as such.

[How to build discord bot with JavaScript #5159 Copyleaks Report.pdf](https://github.com/section-engineering-education/engineering-education/files/7608432/How.to.build.discord.bot.with.JavaScript.5159.Copyleaks.Report.pdf)

**Comment by Kamau-ke** [comment]
@LinusMuema Hello, I made the changes that you requested.

File: content/articles/how-to-build-discord-bot-with-javascript/index.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+![](/education-engineering/how-to-build-discord-bot-with-javascript/yes.png)
+Confirm the creation of the new application and let's proceed.
+3. Here, we will give our bot extra permission to improve its interactivity with users. Click the OAuth2 menu and mark as shown below. Copy the link that appears in the text box.
+   ![](/education-engineering/how-to-build-discord-bot-with-javascript/bot.png)
+   ![](/education-engineering/how-to-build-discord-bot-with-javascript/permission.png)
+4. Open a new tab and paste your link(copied in step 3).
+   ![](/education-engineering/how-to-build-discord-bot-with-javascript/link.png)
+5. select your server, hit authorize.
+   ![](/education-engineering/how-to-build-discord-bot-with-javascript/authorize.png)
+in case you don't have a server. Login to your discord account and create one.
```
Comment (mercymeave) [inline content/articles/how-to-build-discord-bot-with-javascript/index.md:44]: Provide a name to your images for example,    `![image name]/education-engineering/how-to-build-discord-bot-with-javascript/bot.png)`

File: content/articles/how-to-build-discord-bot-with-javascript/index.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+5. select your server, hit authorize.
+   ![](/education-engineering/how-to-build-discord-bot-with-javascript/authorize.png)
+in case you don't have a server. Login to your discord account and create one. 
+### Create New Node.js Project
+First, we need to set up our project, do the following:
+1. Create a folder where you want to store your files.
+2. Initialize a new node.js project, by opening the folder in your text editor.
+3. Install node.js/update if necessary.
+ ### Install dependencies.
+Dependencies are packages that are required by other packages to work. Here we will be using ``discord.js``, ``dotenv`` and ``nodemon``.
```
Comment (mercymeave) [inline content/articles/how-to-build-discord-bot-with-javascript/index.md:51]: Add a space between one heading and the next. This should be done across the article to make it readable.

File: content/articles/how-to-build-discord-bot-with-javascript/index.md [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+```
+```
+npm install discord.js dotenv
+```
+discord.js will allow you to interact with Discord API.
+dotenv will load variables from .env files.
+```
+npm install -D nodemon
+```
+nodemon restart the server every time we make a change to our files.
```
Comment (mercymeave) [inline content/articles/how-to-build-discord-bot-with-javascript/index.md:80]: 1. Provide deeper explanations for these. Yours look like you are taking short notes.
2. Start all sentences with a capital letter.
3. Avoid too much bulleted work and try to explain in full sentences.

File: content/articles/how-to-build-discord-bot-with-javascript/index.md [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+### Building our bot
+First, we need to copy the bot token to allow calls in Discord.js:
+It's done by clicking the copy button in the bot tab. As shown below
```
Comment (mercymeave) [inline content/articles/how-to-build-discord-bot-with-javascript/index.md:83]: I recommend you refrain from using contractions. Do that throughout the article.

File: content/articles/how-to-build-discord-bot-with-javascript/index.md [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+![](/education-engineering/how-to-build-discord-bot-with-javascript/reply.png)
+### Summary
+Here is what we have covered:
+1. Add New Discord Application
+2. Create New Node.js Project
+3. Install Dependencies
+4. Create the bot file
+5. Build our bot
+6. Have bot reply to text
+Below is our entire code:
```
Comment (mercymeave) [inline content/articles/how-to-build-discord-bot-with-javascript/index.md:132]: Avoid bulleted points. Explain these in a paragraph with sentences related to each other.

File: content/articles/how-to-build-discord-bot-with-javascript/index.md [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+});
+client.once("ready", () =>{
+    console.log("BOT IS ONLINE");
+})
+client.login(process.env.TOKEN);
+client.on('message',
+function (messages){
+    if(messages.content.toLocaleLowerCase()==='hello')
+    messages.channel.send('hello' + ' '  + messages.author.username);
+})
```
Comment (mercymeave) [inline content/articles/how-to-build-discord-bot-with-javascript/index.md:150]: Why do you have a code snippet after your summary?

**Review by mercymeave — CHANGES_REQUESTED** [review]
Hello @Kamau-ke 

Thank you for submitting your first article. Here is a list of things to take care of as we get your article ready for publish.

- Avoid using contractions.
- Organize your ideas into sentences to explain the concepts you deliver instead of instructions. Basically, every operation is explained to the user. 
- Use comments for your application to aid in explaining the code.
- Provide a link to the code of the application built.

**Comment by hectorkambow** [comment]
@Kamau-ke 
It seems this PR is still quite a bit aways from being completed. 
 
As a rule of thumb - we only accept content (pull requests) with a complete and fully polished article - ready to be published. 
 
All revisions and edits should be completed on your own forked repo (as to not take up room in the queue). 
 
We will be closing this and please re-submit when it is ready to be published as a new PR (and you can add the duplicate label) and ping the reviewer and myself. 

**Closing a PR does NOT mean the article can not be published (or fixed) it simply means that the edits, revisions, and fixes will happen outside of the queue.**

Looking forward to published your completed and polished article 🚀

**Pull request closed without merging**